File: practice/algorithms/implementation/cats_and_a_mouse/CatsAndMouse.js
'use strict';

const processData = input => {
    const lines = input.split('\n');
    const abms = lines.slice(1).map(line => line.split(' ').map(i => parseInt(i)));

    abms.forEach(abm => {
        const [a, b, m] = abm;

        const catACatchMouseSteps = Math.abs(a - m);
        const catBCatchMouseSteps = Math.abs(b - m);
        if(catACatchMouseSteps == catBCatchMouseSteps) {
            console.log("Mouse C");
        } else if(catACatchMouseSteps < catBCatchMouseSteps) {
            console.log("Cat A");
        } else {
            console.log("Cat B");
        }
    });
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/algorithms/implementation/cats_and_a_mouse/CatsAndMouse.java
import java.util.*;

public class CatsAndMouse {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int tests = Integer.parseInt(stdin.nextLine());
        for(int i = 0; i < tests; i++) {
            String line = stdin.nextLine();
            String[] abmString = line.split(" ");
            int[] abm = new int[3];
            for(int j = 0; j < 3; j++) {
                abm[j] = Integer.parseInt(abmString[j]);
            }
            int a = abm[0];
            int b = abm[1];
            int m = abm[2];
            int catACatchMouseSteps = Math.abs(a - m);
            int catBCatchMouseSteps = Math.abs(b - m);
            if(catACatchMouseSteps == catBCatchMouseSteps) {
                System.out.println("Mouse C");
            } else if(catACatchMouseSteps < catBCatchMouseSteps) {
                System.out.println("Cat A");
            } else {
                System.out.println("Cat B");
            }
        }
        stdin.close();
    }
}